### PR TITLE
feat: Upgrade s3-object-lambda to CDK v2 and fix #787

### LINKED ISSUE #787: Build of s3-object-lambda fails
### Describe the bug

Build of s3-object-lambda fails: ```lib/s3-object-lambda-stack.ts:92:15 - error TS2322: Type '{ AwsLambda: { FunctionArn: string; }; }' is not assignable to type 'IResolvable | ContentTransformationProperty'. Object literal may only specify known properties, and ''AwsLambda'' does not exist in type 'IResolvable | ContentTransformationProperty'.```

This seems to be caused by casing changes in the underlying library: AwsLambda -> awsLambda.

### Expected Behavior

Build should not fail.

### Current Behavior

Build fails.

### Reproduction Steps

Run a build

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

latest

### Framework Version

_No response_

### Node.js Version

all

### OS

all

### Language

Typescript

### Language Version

_No response_

### PR DESCRIPTION
**Pull request by markusl**
Fixes #787

Use CDK v2 and use correct casing for `contentTransformation` properties.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: typescript/s3-object-lambda/lib/s3-object-lambda-stack.ts [hunk 1]
```diff
@@ -24,85 +27,87 @@ export class S3ObjectLambdaStack extends cdk.Stack {
...
-              'AwsLambda': {
-                'FunctionArn': `${retrieveTransformedObjectLambda.functionArn}`
-              }
+      name: OBJECT_LAMBDA_ACCESS_POINT_NAME,
+      objectLambdaConfiguration: {
+        supportingAccessPoint: accessPoint,
+        transformationConfigurations: [{
+          actions: ['GetObject'],
+          contentTransformation: {
+            'awsLambda': {
```
Comment (peterwoodworth) [inline typescript/s3-object-lambda/lib/s3-object-lambda-stack.ts:97]: Can we revert the casing please? This was a breaking change that will get patched on the next release

**Review by peterwoodworth — DISMISSED** [review]
Thanks for the upgrade to v2, I had thought we converted everything to v2 already but we must have missed this one.

File: typescript/s3-object-lambda/lib/s3-object-lambda-stack.ts (hunk 1 above)
Comment (markusl) [reply typescript/s3-object-lambda/lib/s3-object-lambda-stack.ts]: Sure, it's now back to previous casing

**Comment by peterwoodworth** [comment]
@merfigyio refresh

**Comment by peterwoodworth** [comment]
@Mergifyio refresh

**Comment by peterwoodworth** [comment]
@Mergifyio update

**Review by peterwoodworth — APPROVED** [review]

**Pull request merged**